### ISSUE TITLE
Sync command in examples/client_test.cpp not working as expected

### ISSUE THREAD

**Issue by amd128** [bittorrent/scout#21]
In the ses.synchronize() call, the finalize callback has a for loop that looks like it should print each entry to cout after updates to the vector have been received from the DHT. However, when I use the sync command, no entries (not even the one I just created) get printed to cout. I have added print lines to verify that the finalize callback is actually getting called, and it appears as though there are no entries being stored in the vector anymore by that point.

**Comment by ssiloti** [comment]
Please post the sequence of commands you are using.

**Comment by amd128** [comment]
```
PS C:\Users\AMD\Documents\GitHub\scout\examples\bin\msvc-14.0\debug\link-static\threading-multi> .\client_test.exe gen-key key1.txt
PS C:\Users\AMD\Documents\GitHub\scout\examples\bin\msvc-14.0\debug\link-static\threading-multi> .\client_test.exe gen-key key2.txt
PS C:\Users\AMD\Documents\GitHub\scout\examples\bin\msvc-14.0\debug\link-static\threading-multi> .\client_test.exe dump-key key2.txt
Public key: fdfd28d18169227f8be4be448070e95573b8e34774991c265bd988b388deae64
PS C:\Users\AMD\Documents\GitHub\scout\examples\bin\msvc-14.0\debug\link-static\threading-multi> .\client_test.exe sync .\key1.txt fdfd28d18169227f8be4be448070e95573b8e34774991c265
bd988b388deae64 42 "Testing testing testing" 2> sync_err.txt
```

Basically, nothing gets printed to COUT, only CERR. But, from what I can tell, it looks like the for loop in the finalize callback should at least be printing "42 Testing testing testing" to COUT.

**Comment by ssiloti** [comment]
It works for me, what output do you see on stderr?

**Comment by amd128** [comment]
It is quite a lot of output, almost 1000 lines.
[sync_err.txt](https://github.com/bittorrent/scout/files/794529/sync_err.txt)

**Comment by ssiloti** [comment]
It looks like the value you specified didn't get passed through to the synchronize function. Make sure you're using the latest master version. Revisions prior to 537d560c4fe8aa25ef7359fb3a9580f28a11f94a have a race condition which can cause the entries vector to be cleared.

**Comment by amd128** [comment]
Yep that was my problem. Sorry about that and thanks for the help!

**Comment by amd128** [comment]
I am now using the latest master version, but when I call sync it doesn't stop running. I am getting the expected output to standard out, but I have to press ctrl+c to stop client_test.exe from running forever. Same command sequence as earlier, but here's what's getting written to standard error: 
[sync_err.txt](https://github.com/bittorrent/scout/files/800112/sync_err.txt)

**Comment by c7huang** [comment]
I encountered the same problem while simulating two peers on the same operating system. This command might have to be tested on different computers.

**Comment by amd128** [comment]
My problem is slightly different from what hcj1009 experienced, as I am using the same private key every time.  

Also, I tried deleting dht.dat and the two .log files before calling the sync command, but that hasn't helped. I still have to use ctrl+c to get the program to stop running.
